### PR TITLE
go.mod: add newly created go.mod

### PR DESCRIPTION
**Pull request by Asalle**
#16

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module github.com/roc-project/roc-go
```
Comment (gavv) [inline go.mod:1]: Shouldn't we place this file into `roc` and rename the module to `github.com/roc-project/roc-go/roc`?

File: go.mod [hunk 2]
```diff
@@ -0,0 +1,3 @@
...
+
+go 1.14
```
Comment (gavv) [inline go.mod:3]: We actually support at least go 1.12 and it's even checked in our travis build.

File: go.mod [hunk 3]
```diff
@@ -0,0 +1,3 @@
+module github.com/roc-project/roc-go
+
```
Comment (gavv) [inline go.mod:2]: Shouldn't we also add GO111MODULE=on to `env` section in travis?

File: go.mod (hunk 1 above)
Comment (Asalle) [reply go.mod:1]: I'm not sure why do want to do this? do we plan to have more folders in `github.com/roc-project/roc-go/` with their own `go.mod`s? or is it just for the sake of changing the module name to `roc`?

File: go.mod (hunk 3 above)
Comment (Asalle) [reply go.mod:2]: alright

File: go.mod (hunk 2 above)
Comment (Asalle) [reply go.mod:3]: alright

File: go.mod (hunk 3 above)
Comment (Asalle) [reply go.mod:2]: I'm wondering how did the ci pass :thinking:

Comment (gavv) [reply go.mod:2]: Hm, mys suggestion was not quite correct. By just adding a line to `env` we tell travis to build all combinations of these three exports:

```
  - GOTESTFLAGS=""
  - GOTESTFLAGS="-race"
  - GO111MODULE=on
```

image

We should use `global`: https://docs.travis-ci.com/user/environment-variables/#global-variables

File: go.mod (hunk 1 above)
Comment (gavv) [reply go.mod:1]: My bad, my understanding of go modules was wrong. I though that one module == one package, but that's not true. Let's keep it as is.

Comment (Asalle) [reply go.mod:1]: alright!

File: go.mod (hunk 3 above)
Comment (Asalle) [reply go.mod:2]: fixed

Comment (gavv) [reply go.mod:2]: The build matrix is broken now for some reason.

Was: https://travis-ci.org/github/roc-project/roc-go/builds/699219954

image

Now: https://travis-ci.org/github/roc-project/roc-go/builds/699428591

image

Comment (gavv) [reply go.mod:2]: Maybe try to remove `- ` from `- global:` and `- jobs`.

Comment (gavv) [reply go.mod:2]: https://config.travis-ci.com/explore

Comment (Asalle) [reply go.mod:2]: I'll take a look into it

Comment (Asalle) [reply go.mod:2]: yes, don't know what I was thinking

**Comment by gavv** [comment]
Forgot one more detail, please also add GO111MODULE=on to Makefile, like here: https://github.com/gavv/webrtc-cli/blob/master/Makefile#L1-L2

**Comment by gavv** [comment]
Sorry, one more concern.

It seems that moving go.mod to roc/ still makes sense. When we'll decide to tag v2 version (or v3, or v4, ...), we'll have to append it to go module name in go.mod, like:

```
module github.com/gavv/roc-go/v2
```

If go.mod is in project root, the user import will look like:

```
import "github.com/gavv/roc-go/v2/roc"
```

If, however, go.mod is in roc/ sub-directory, the user import would be:

```
import "github.com/gavv/roc-go/roc/v2"
```

The second form is more familiar, in my opinion. For most go packages, you just append the version to the import path. The first form is usually used only for multi-package repositories, however our repo is single-package.

Thus, I'd prefer the second form for roc-go. So maybe move go.mod to roc/ directory after all, what do you think?

**Comment by Asalle** [comment]
@gavv then, maybe it makes sense to remove the folder and just put all the files in the project root? like here https://github.com/alicebob/miniredis

**Comment by gavv** [comment]
> then, maybe it makes sense to remove the folder and just put all the files in the project root?

Technically possible, but in this case another inconsistency appears: the repo and the import path will be ".../roc-go", but the package name will be "roc". Which is allowed but considered bad practice. (Probably we already discussed it... Don't remember.)

**Comment by gavv** [comment]
I've renamed github org (see mailing list), so we also need to update go.mod accordingly.

**Comment by Asalle** [comment]
> I've renamed github org (see mailing list), so we also need to update go.mod accordingly.

cool, will change

alright, then let's make it `roc` with go.mod inside the folder

File: roc/go.mod [hunk 4]
```diff
@@ -0,0 +1,3 @@
+module github.com/roc-streaming/roc-go
```
Comment (gavv) [inline roc/go.mod:1]: Well it should be `github.com/roc-streaming/roc-go/roc` then? :)

Comment (Asalle) [reply roc/go.mod]: that's right

**Pull request merged**